pkg/kubectl/cmd/util/factory.go

```diff
@@ -197,32 +199,57 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clients: clients,
 		flags:   flags,
 
-		Object: func() (meta.RESTMapper, runtime.ObjectTyper) {
+		// If discoverDynamicAPIs is true, make API calls to the discovery service to find APIs that
+		// have been dynamically added to the apiserver
+		Object: func(discoverDynamicAPIs bool) (meta.RESTMapper, runtime.ObjectTyper) {
 			cfg, err := clientConfig.ClientConfig()
 			CheckErr(err)
 			cmdApiVersion := unversioned.GroupVersion{}
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			CheckErr(err)
+			if discoverDynamicAPIs {
+				client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+				CheckErr(err)
+				
+				versions, gvks, err := GetThirdPartyGroupVersions(client.Discovery())
+				CheckErr(err)
+				if len(versions) > 0 {
+					preferredExternalVersion := versions[0]
+
+					accessor := meta.NewAccessor()
+					thirdPartyMapper, err := kubectl.NewThirdPartyResourceMapper(versions, gvks)
+					CheckErr(err)
+					groupMeta := apimachinery.GroupMeta{
+						GroupVersion:  preferredExternalVersion,
+						GroupVersions: versions,
+						RESTMapper:    thirdPartyMapper,
+						SelfLinker:    runtime.SelfLinker(accessor),
+						InterfacesFor: func (version unversioned.GroupVersion) (*meta.VersionInterfaces, error) {
+							for ix := range versions {
+								if versions[ix].String() == version.String() {
+									return &meta.VersionInterfaces{
+										ObjectConvertor: thirdpartyresourcedata.NewThirdPartyObjectConverter(api.Scheme),
+										MetadataAccessor: accessor,
+									}, nil
+								}
+							}
+							return nil, fmt.Errorf("unsupported storage version: %s (valid: %v)", version, versions)
+						},
+					}
 
-			outputRESTMapper := kubectl.OutputVersionMapper{RESTMapper: mapper, OutputVersions: []unversioned.GroupVersion{cmdApiVersion}}
-
-			// eventually this should allow me choose a group priority based on the order of the discovery doc, for now hardcode a given order
```

Comment:
Can't eliminate the `PriorityRESTMapper`.  It is what is providing automatic disambiguation of conflicts for known priorities.